tests/test_authorisation.py

```diff
@@ -351,9 +340,136 @@ def test_different_user_levels_have_correct_flagging_permissions(
                     "assess/flag/app123",
                     follow_redirects=True,
                 )
+                assert False, "Did not throw expected error."
             except Exception as e:
                 # Redirect to authenticator
                 assert (
                     e.args[0]
                     == "Following external redirects is not supported."
                 )
+
+    @pytest.mark.parametrize(
+        "claim,expect_resolve_flag_available",
+        [
+            (test_commenter_claims, False),
+            (test_assessor_claims, False),
+            (test_lead_assessor_claims, True),
+        ],
+    )
```

Comment:
There is a boolean trap here, with some logic built-in to the test

My preference is usually to split these into two separate tests, rather than maintain the logic

I'm not too fussed about this, but let me know what you think

Reply:
Im not too concerned about it as it's test code (i agree diff story in prod code), the reason I have parametrised over writing individual tests it to keep the line count down and keep the test files readable and extendable if we want to test for more roles in the future (or switch the permissions of certain roles). I think it would be easier to go in and switch the bool rather than rewrite each test.

Reply:
Yeah reading the full test in isolation I actually like the way you've done it.  I think if we had separate tests we would still parametrize the claims, but the boolean logic would just be two separate tests so it would be shuffling from one tests parametrized claims to anothers - but let's leave it how it is cause it's not a big deal and looks good as it is just now.